Movalpolos region NM audit

CriticalXI opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

Adds several NMs to the Movalpolos Region:
- Swashstox Beadblinker
- Goblin Collector
- Sword Sorcerer Solisoq

Several other NMs have had proper behavior/mechanics introduced to them. Droplists have been adjusted according to the TH chart.

## Steps to test these changes

See list of changed NMs in change list, pop the NMs in game and ensure their logic triggers correctly.

CriticalXI committed:
Movalpolos region NM audit

CriticalXI committed:
New line at end of file

Xaver-DaRed commented on scripts/globals/mobskills/flying_hip_press.lua:
```diff
@@ -10,18 +10,25 @@
+     -- Bugbear Matman has stronger Flying Hip Press
+    if mob:getID() == newtonID.mob.BUGBEAR_MATMAN then
+        bonus = 700
+    end
```
I really dont like this method of adjusting damage.
What I mean is this:
If something is modifying the TP move dmg, it should be handled via modifier IMO.

thoughts?

TeoTwawki replied:
the pattern we've been using since dsp is to conditional on `getPool()` since mob ID can (albeit rarely) change and we'd wind up with a lot of unique one time used modifiers making it messy.

example: https://github.com/LandSandBoat/server/blob/3dfe1f76e9a5c4fe4392de6ffb77ea2bdc2fb4ef/scripts/globals/mobskills/restoral.lua#L22

TeoTwawki replied:
In general, SE _usually but not always_ has a 2nd copy of  the mobskill in the dats that we can rename to make alternates with. A capture will tell us if is the same IID the normal mobs use or if the NM has such an alternate on retail. In some cases the difference really comes down  to the NM simply being that much stronger than the normal mobs making its TP skill stronger but in most cases they just devote a 2nd dat entry to it. Past iteration of the project have not implemented most of that and have been doing conditional branches for most skills instead. If is duplicate ID is used on retail, I would readily accept that in a pr. else I'd go for getPool method.

CriticalXI replied:
Well, I'll have to look into this for quite a few NMs I've corrected on Wings. This has been my goto method for introducing different mechanics that NMs have for normal TP moves (Cemetery Cherry comes to mind). I'll ask for a capture.

TeoTwawki replied:
> for introducing different mechanics that NMs have for normal TP moves 

Any that are only a power change, go for getPool anyway if they animate identically. getPool is safe. more complex logic changes you'll want that separate scrip if the way SE did them allows for it.

CriticalXI replied:
I received a capture for this NM, and its the same ID for both skill and animation. So I've adjusted the line to use mob:getPool() instead.

I am running into a bit of an issue with Flying Hip Press actually. It seems the `bonus` variable is actually a cap on the amount of damage a breath move can do, and uncapped this ability is doing 4000~ damage. So while breath moves should do less damage the lower the mob gets, its going from 4000 down to 1000 but since I've set the cap to 700, you will see zero change to the damage flying hip press does all the way down until the mob dies. It always does exactly 700 damage with no variance.

Xaver-DaRed commented on scripts/zones/Newton_Movalpolos/mobs/Goblin_Collector.lua:
```diff
@@ -0,0 +1,27 @@
+    mob:removeListener("COLLECTOR_ATTACK")
+end
+
+return entity
```
add newline at the end of the file pls

Xaver-DaRed commented on scripts/zones/Newton_Movalpolos/mobs/Goblin_Swordsman.lua:
```diff
@@ -0,0 +1,19 @@
+    xi.mob.phOnDespawn(mob, ID.mob.SWASHSTOX_BEADBLINKER_PH, 15, 10800)
+end
+
+return entity
```
newline here, pls

Xaver-DaRed commented on scripts/zones/Newton_Movalpolos/mobs/Swashstox_Beadblinker.lua:
```diff
@@ -1,13 +1,57 @@
+    end
+end
+
+return entity
```
newline at the end needed

Xaver-DaRed commented on scripts/zones/Newton_Movalpolos/npcs/qm_collector.lua:
```diff
@@ -0,0 +1,17 @@
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+    if npcUtil.tradeHas(trade, xi.items.PREMIUM_BAG) and npcUtil.popFromQM(player, npc, ID.mob.GOBLIN_COLLECTOR) then
```
Use `npcUtil.tradeHasExactly` unless the trade actually works with other items in it, so long as the premium bag is in it.
If it does, please add a comment so we know it's supposed to do that.

CriticalXI committed:
New lines and tradeHasExactly correction

CriticalXI commented on scripts/zones/Oldton_Movalpolos/mobs/Goblin_Wolfman.lua:
```diff
@@ -10,6 +10,20 @@ local entity = {}
+        mob:addMod(xi.mod.ATTP, 160)
+        mob:setLocalVar("weaponOn", 1)
+    elseif not mob:hasStatusEffect(xi.effect.BLOOD_WEAPON) and weaponOn == 1 then
+        mob:delMod(xi.mod.DELAY, 0)
```
Note for myself, need to correct the delMod to actually delete the delay/attack set during blood weapon.

TeoTwawki commented on scripts/zones/Newton_Movalpolos/mobs/Bugbear_Matman.lua:
```diff
@@ -6,6 +6,17 @@ require("scripts/globals/hunts")
+
+entity.onMobWeaponSkillPrepare = function(mob,target)
+    -- Below 30% Bugbear Matman heavily prefers Heavy Whisk
+	if mob:getHPP() <= 30 and math.random() > 0.4 then
```
sneaky tab slipped in here. most editors have editorconfig support if you want conversion to be automated. Our project has a file that tells editorconfig enhances editor (and even git itself) what formatting to use.

TeoTwawki commented on scripts/zones/Newton_Movalpolos/mobs/Swashstox_Beadblinker.lua:
```diff
@@ -1,13 +1,57 @@
 -----------------------------------
 -- Area: Newton Movalpolos
 --   NM: Swashstox Beadblinker
------------------------------------
```
We want to avoid arbitrarily changing the length of these

TeoTwawki commented on sql/mob_pools.sql:
```diff
@@ -1697,7 +1697,7 @@ INSERT INTO `mob_pools` VALUES (1641,'Goblin_Bouncer','Goblin_Bouncer',133,0x000
 INSERT INTO `mob_pools` VALUES (1643,'Goblin_Butcher','Goblin_Butcher',133,0x0000EF0100000000000000000000000000000000,1,1,4,240,100,0,1,0,1,0,0,0,8067,131,0,0,0,0,0,133,133);
 INSERT INTO `mob_pools` VALUES (1644,'Goblin_Chapman','Goblin_Chapman',133,0x0000B30200000000000000000000000000000000,4,4,12,240,100,0,1,0,1,0,0,0,835,131,0,0,2,0,0,133,133);
-INSERT INTO `mob_pools` VALUES (1645,'Goblin_Collector','Goblin_Collector',133,0x0000BC0200000000000000000000000000000000,3,3,11,220,100,0,1,0,1,2,0,0,0,3,0,0,1,0,0,133,133);
+INSERT INTO `mob_pools` VALUES (1645,'Goblin_Collector','Goblin_Collector',133,0x0000BC0200000000000000000000000000000000,3,3,11,150,200,0,1,0,1,2,17,0,0,3,0,0,463,0,0,1166,133);
```
The "fast attacks" mentioned on wiki appear to be bogus. I just checked him seemed to be the normal 240 most mobs get except that he has a high rate of triple attack (more consistent with a thf main, which makes sense when you realize mobs don't really do the half-level sub thing the way players do they get full traits of both jobs).

when an entity attacks the combat round timer counts from the begging, so you can start your next attack as the 3rd hit end. My thf on retail can look like hundred fists with some haste and my triple attacks going off.

fortunately, I don't need to record this because somebody already did so you can see the 3 attacks with a very brief pause: https://youtu.be/muuYcCtqmXo?t=27

CriticalXI replied:
I've corrected his job to THF and removed the lowered delay. Also realized there is a line that is said when spawning him so added that as well. Added a TODO to eventually find the jumping animation he does on spawn.

TeoTwawki replied:
I don't know that he is a thf main - just that he has that much triple attack. I don't know if it being his main or his sub even matters, but given that both wiki's say when there is a high probability someone in the future will see that and come back to "fix" his job here, even if  turns out its right (we just don't know yet).

Given that, I think we should play it safe and use `3,6` rather than `6,3` here for the time being.

(on retail, main or sub it'd be the same traits/amounts)

Xaver-DaRed commented:
Since your other PRs got merged, this needs a rebase

Xaver-DaRed commented on scripts/zones/Newton_Movalpolos/mobs/Goblin_Collector.lua:
```diff
@@ -0,0 +1,27 @@
+    -- Resets threat on every auto attack
+    mob:addListener("ATTACK","COLLECTOR_ATTACK", function(mob)
+        mob:resetEnmity(target)
+    end)
```
From CI, you are " shadowing upvalue argument mob on line 16"
```
    mob:addListener("ATTACK","COLLECTOR_ATTACK", function(goblin)
        goblin:resetEnmity(target)
    end)
```
Use this instead, or change "goblin" to whatever else you see fit

CriticalXI committed:
Further adjustments to Movalpolos NMs

CriticalXI committed:
Goblin Collector adjustments

CriticalXI committed:
Remove unused ID from flying hip press

TeoTwawki commented on scripts/zones/Newton_Movalpolos/IDs.lua:
```diff
@@ -22,15 +22,23 @@ zones[xi.zone.NEWTON_MOVALPOLOS] =
+            [16826507] = 16826517, -- 92.145, 15.500, 66.595
+            [16826514] = 16826517, -- 88.412, 15.421, -19.950
+        },
+        SWASHSTOX_BEADBLINKER    = 16826510,
```
Since there is actually 2 copies of this, just like leaping lizzy in south gusta, I would expect him to have a PH for each ID and function similarly - with one of the above to PH entries having this ID for what the spawn rather than it on a separate line. But I've never camped this guy that I can remember - does he work differently?
```
16826507	Goblin Swordsman
16826508	Moblin Aidman
16826509	Moblin Engineman
16826510	Swashstox Beadblinker

16826514	Goblin Swordsman
16826515	Moblin Aidman
16826516	Moblin Engineman
16826517	Swashstox Beadblinker
```

CriticalXI commented:
* Merge conflicts resolved
* Adjusted Goblin Collecter back to WHM/THF with Triple Attack
* Adjusted all droplists to use the new variable for drop rate
* Adjusted Swashstox Beadblinker to use both spawns listed in mob_spawn_points

TeoTwawki reviewed (approved):
everything I noted has been adjusted, nothing else stands out to me
